src/Nest/ElasticClient-Reindex.cs

```diff
@@ -14,5 +14,11 @@ public IObservable<IReindexResponse<T>> Reindex<T>(Func<ReindexDescriptor<T>, Re
 			var observable = new ReindexObservable<T>(this, _connectionSettings, reindexDescriptor);
 			return observable;
 		}
+
+		/// <inheritdoc />
+		public IObservable<IReindexResponse<IDocument>> Reindex(Func<ReindexDescriptor<object>, ReindexDescriptor<object>> reindexSelector)
```

Comment:
To be truly lazy `IDocument` this signature needs to change to `Reindex(Func<ReindexDescriptor<IDocument>, ReindexDescriptor<IDocument>> reindexSelector)`